src/epcpm/parameterstobitfieldsc.py

```diff
@@ -0,0 +1,504 @@
+                        f".length = {member.bit_length},",
+                        f".item = &interfaceItem_{epcpm.pm_helper.convert_uuid_to_variable_name(member.parameter_uuid)},",
+                    ],
+                    f"}},",
+                ]
+                for i, member in enumerate(members)
+            ],
+            f"}};",
+            "",
+        ]
+
+    def gen_members_interface(self) -> typing.List[str]:
+        """
+        Generate the static modbus bitfield member additional definitions.
+
+        Returns:
+            list: bitfield member additional definitions
+        """
+        name_uuid = epcpm.pm_helper.convert_uuid_to_variable_name(
+            self.wrapped.parameter_uuid
+        )
+        members = self.wrapped.children
+        array_name = f"staticmodbusBitfieldItems_{name_uuid}"
+
+        return [
+            f".staticmodbusMembers = {array_name},",
+            f".staticmodbusMembersCount = {len(members)},",
+        ]
+
+
+@builders(epcpm.sunspecmodel.DataPointBitfield)
+@attr.s
+class DataPointBitfield:
+    wrapped = attr.ib()
+    parameter_uuid_finder = attr.ib()
+
+    def gen(self) -> typing.List:
+        """
+        Generate a SunSpec bitfield variable and bitfield member extern definitions.
+        Returns:
+            list: bitfield variable and bitfield member extern definitions (.c)
+            list: bitfield variable and bitfield member extern definitions (.h)
+        """
+        name_uuid = epcpm.pm_helper.convert_uuid_to_variable_name(
+            self.wrapped.parameter_uuid
+        )
+        members = self.wrapped.children
+        array_name = f"sunspecBitfieldItems_{name_uuid}"
+
+        c_lines = []
+        h_lines = []
+
+        for member in members:
+            member_name_uuid = epcpm.pm_helper.convert_uuid_to_variable_name(
+                member.parameter_uuid
+            )
+            variable_name = f"interfaceItem_variable_{member_name_uuid}"
+            variable_type = epcpm.parameterstointerface.sunspec_types[
+                self.parameter_uuid_finder(member.type_uuid).name
+            ]
+
+            h_lines.append(
+                f"extern {variable_type} {variable_name};",
+            )
+
+            # TODO: initializer that doesn't assume int16_t
+            c_lines.append(
+                f"{variable_type} {variable_name} = 0;",
+            )
+
+        c_lines.append("")
+        h_lines.append("")
+
+        h_lines.extend(
+            [
+                f"extern InterfaceItem_BitfieldMember {array_name}[{len(members)}];",
+            ]
+        )
+
+        return c_lines, h_lines
+
+    def gen_common_interface(self) -> typing.List[str]:
+        """
+        Generate a SunSpec bitfield common definitions for the interface item.
+
+        Returns:
+            list: bitfield common definitions
+        """
+        bits_per_modbus_register = 16
+        bit_length = bits_per_modbus_register * self.wrapped.size
```

Comment:
should this be renamed to bitfield\_bit\_length?

Reply:
No, I used the variable name `bit_length` to match other code that uses the same name. Consistency.

Reply:
Changed in 7c020f4c2743bae15875da13eb0486b3427ac3a5

Also removed some unused code that was somewhat related.

Reply:
i think there are other mentions of bit\_length.  Should they all be change bitfield\_bit\_length?

Reply:
No, the other `bit_length` variables have to do with the length of bitfield members and those are consistently using `bit_length` in several places. The ones I just changed have to do with the bitfield length (16 or 32 in this case).